inc/functions.php

```diff
@@ -87,3 +87,52 @@ function core_sitemaps_get_max_urls( $type = '' ) {
 	 */
 	return apply_filters( 'core_sitemaps_max_urls', CORE_SITEMAPS_MAX_URLS, $type );
 }
+
+if ( ! function_exists( 'esc_xml' ) ) :
+/**
+ * Escaping for XML blocks.
+ *
+ * @since 5.5.0
+ *
+ * @param string $text
+ * @return string
+ */
+function esc_xml( $text ) {
+	$safe_text = wp_check_invalid_utf8( $text );
+	$safe_text = _wp_specialchars( $safe_text, ENT_QUOTES );
+	$safe_text = html_entity_decode( $safe_text, ENT_HTML5 );
```

Comment:
The reason for these new `esc_xml()` and `esc_xml__()` functions is that far too many developers think that since they can use `&amp;`, `&apos;`, &quot;`, etc in XML then they can **also** use all the named character references they are used to using in HTML, e.g., `&nbsp;`, `&hellip;`, etc...but they cannot...and doing so will result in a non-well-formed XML instance.

The call to [html_entity_decode( $safe_text, ENT_HTML5 )](https://www.php.net/manual/en/function.html-entity-decode.php) will replace all of the named character references defined in the HTML spec with their equivalent Unicode code points (e.g. `&nbsp;` will become `\xA0`, etc).

It would be nice PHP had a native function that would replace them with character references (e.g., '`&nbsp;' would become `&#A0;`) but unfortunately it doesn't :-(

Note that all uses of `esc_attr()` in this plugin (e.g., in `Core_Sitemaps_Renderer::get_sitemap_xml()`) should be replaced with calls to `esc_xml()`, but I thought it best to keep this PR strictly related to the stylesheet.  Will do another PR for that.

Reply:
You can see what happens when content like `&nbsp;` is included in an XML instance by doing the following (in v0.2.0 of this plugin...i.e., without the changes in this PR) :

```
add_filter( 'core_sitemaps_posts_url_list', function( $url_list ) {
   foreach ( $url_list as &$url_item ) {
      $url_item['foo'] = 'This&nbsp;will be a non-well-formed sitemap, and it will fail to render in the browser';
   }

   return $url_list;
} );
```

* Chrome will just show a blank screen (and no error message in the console)
* Firefox will show an error screen (without any error message), but will show `XML Parsing Error: undefined entity` in the console)
* Edge and IE will show a screen with the text of content of each element in the sitemap (which is not conformant with the XML spec, but it's Microsoft, so what do you expect), and will display `Invalid tag start: "<?". Question marks should not start tags.` in the console, which is not actually what the error is, but as we all know when parse errors occur it can be hard to output the correct error message)
* Not sure what Safari, Opera, etc will do, but I expect one of the above

Would anyone explicitly include `&nbsp;`, `&hellip;`, etc in element content in a sitemap?  Maybe not, but they very well could include content stored in post meta, which easily could contain HTML named character references (since such post meta was likely stored so that it could be displayed in HTML).